### DISCUSSION TITLE
Vaultwarden Update Script Error

### DISCUSSION THREAD

**Discussion by draxunon** [tteck/Proxmox#3522] [Help Needed]
Receiving the following error when running the upgrade script.

` / Updating VaultWarden to 1.32.0 (Patience)   
[ERROR] in line 84: exit code 0: while executing command cargo build --features "sqlite,mysql,postgresql" --release &> /dev/null
`

I have reviewed previous discussions indicating this error is typically resolved by adding resources. My LXC container has 6vCPU and 4GB RAM.

![image](https://github.com/user-attachments/assets/71a91de0-825a-4372-b201-2c9d664ce40f)

Also space is not an issue.
<img width="462" alt="image" src="https://github.com/user-attachments/assets/38baba7a-8658-48aa-a8dd-398105f11437">

When I manually run the cargo build command this is the output. Any ideas?

`root@vw1:~/vaultwarden# cargo build --features "sqlite,mysql,postgresql" --release
   Compiling mysqlclient-sys v0.4.0
   Compiling ref-cast v1.0.23
   Compiling chrono-tz-build v0.3.0
   Compiling devise v0.4.1
   Compiling pest_meta v2.7.11
   Compiling psm v0.1.21
error: failed to run custom build command for `mysqlclient-sys v0.4.0`

Caused by:
  process didn't exit successfully: `/root/vaultwarden/target/release/build/mysqlclient-sys-6242426b274fb897/build-script-build` (exit status: 101)
  --- stdout
  cargo::rerun-if-env-changed=MYSQLCLIENT_VERSION
  cargo::rerun-if-env-changed=MYSQLCLIENT_INCLUDE_DIR
  cargo::rerun-if-env-changed=MYSQLCLIENT_INCLUDE_DIR_X86_64_UNKNOWN_LINUX_GNU
  cargo::rerun-if-env-changed=MYSQLCLIENT_LIB
  cargo::rerun-if-env-changed=MYSQLCLIENT_LIB_DIR
  cargo::rerun-if-env-changed=MYSQLCLIENT_LIB_DIR_X86_64_UNKNOWN_LINUX_GNU
  cargo::rerun-if-env-changed=MYSQLCLIENT_LIBNAME
  cargo::rerun-if-env-changed=MYSQLCLIENT_LIBNAME_X86_64_UNKNOWN_LINUX_GNU
  cargo::rerun-if-env-changed=MYSQLCLIENT_STATIC
  cargo::rerun-if-env-changed=MYSQLCLIENT_VERSION_X86_64_UNKNOWN_LINUX_GNU
  cargo::rerun-if-env-changed=MYSQLCLIENT_LIB_X86_64_UNKNOWN_LINUX_GNU
  cargo::rerun-if-env-changed=MYSQLCLIENT_STATIC_X86_64_UNKNOWN_LINUX_GNU
  cargo:rerun-if-env-changed=MYSQLCLIENT_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=MYSQLCLIENT_STATIC
  cargo:rerun-if-env-changed=MYSQLCLIENT_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=LIBMARIADB_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=LIBMARIADB_STATIC
  cargo:rerun-if-env-changed=LIBMARIADB_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR

  --- stderr
  thread 'main' panicked at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/mysqlclient-sys-0.4.0/build.rs:89:5:
  Did not find a compatible version of libmysqlclient.
  Ensure that you installed one and teached mysqlclient-sys how to find it
  You have the following options for that:

  * Use `pkg_config` to automatically detect the right location
  * Use vcpkg to automatically detect the right location.
  You also need to set `MYSQLCLIENT_VERSION` to specify which
  version of libmysqlclient you are using
  * Set the `MYSQLCLIENT_LIB_DIR` and `MYSQLCLIENT_VERSION` environment
  variables to point the compiler to the right directory and specify
  which version is used
  * Make the `mysql_config` binary avaible in the environment that invokes
  the compiler
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...`

**Comment by TheTaran** [comment]
I ahve the same error when i want to update!

**Comment by tteck** [comment]
https://github.com/tteck/Proxmox/issues/3515

**Reply by TheTaran** [reply]
Thanks, solved. Problem was out of disk space.

**Reply by draxunon** [reply]
As mentioned in my original post space is not an issue the / mount has 20G of available space. 

As mentioned in the #3515 I ran the commands manually to get output from the build command as is listed above.  This is on a Debian 11 build of the LXC as it was build before the script upgraded to Debian 12 base container.  The script to upgrade was run before and after fully updating the underlying OS.

**Comment by nicolouts** [comment]
Hi all,
I had the same issue and I try all the solution provide in some discussion.
But I always had error on line 84 (cargo build .....)
I connect to my admin vaultwarden GUI and I see that I my database was sqlite.
So I follow the script manually to do all step until line 84
I change the command at line 84 
     "cargo build --features "sqlite,mysql,postgresql" --release &>/dev/null"
by
     "cargo build --features "sqlite" --release &>/dev/null"

sqlite only and I was able to complete the upgrade.
Good luck to you

Nicolouts
